Description: Test for all database dumping programs that postgresql uses,
 to avoid breaking backup-manager on a broken postgresql installation.
Author: Georgios M. Zarkadas <gz@member.fsf.org>
Forwarded: https://github.com/sukria/Backup-Manager/pull/16
Last-Update: 2011-12-25

--- a/lib/backup-methods.sh
+++ b/lib/backup-methods.sh
@@ -892,8 +892,10 @@
     debug "backup_method_pgsql ($method)"
 
     info "Using method \"\$method\"."
-    if [[ ! -x $pgdump ]]; then
-        error "The \"pgsql\" method is chosen, but \$pgdump is not found."
+    if [[ -x $pgdump ]] && [[ -x ${pgdump}all ]]; then
+        :
+    else
+        error "The \"postgresql\" method is chosen, but \$pgdump and/or \$pgdumpall are not found."
     fi
 
     # Allow empty host when connecting to postgress with unix sockets.
